services/localfile/server/localfile.go

```diff
@@ -774,6 +796,50 @@ func (s *server) Symlink(ctx context.Context, req *pb.SymlinkRequest) (*emptypb.
 	return &emptypb.Empty{}, nil
 }
 
+func (s *server) Mkdir(ctx context.Context, req *pb.MkdirRequest) (_ *emptypb.Empty, retErr error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	recorder := metrics.RecorderFromContextOrNoop(ctx)
+
+	if req.GetDirDescription() == nil {
+		recorder.CounterOrLog(ctx, localfileMkdirFailureCounter, 1, attribute.String("reason", "missing_dst"))
+		return nil, status.Error(codes.InvalidArgument, "destination must be filled in")
+	}
+	d := req.GetDirDescription()
+
+	a := d.GetAttrs()
+	if a == nil {
+		recorder.CounterOrLog(ctx, localfileMkdirFailureCounter, 1, attribute.String("reason", "missing_attrs"))
+		return nil, status.Errorf(codes.InvalidArgument, "must send attrs")
+	}
+	filename := a.Filename
+	logger.Info("create directory", filename)
+	dir, _, err := setupOutputDir(a)
```

Comment:
Looks like you're ignoring the immutable bit. We should either support it or explicitly fail when we see it.

Reply:
I think adding the immutable to a directory when we create it will leads to the result that no files can be created, renamed or deleted. Then there is not point in creating the directory. So let me explicitly fail when see it.